lib/Parser/RdfXml.php

```diff
@@ -800,13 +800,19 @@ public function parse($graph, $data, $format, $baseUri)
         $this->initXMLParser();
 
         /* parse */
-        if (!xml_parse($this->xmlParser, $data, false)) {
-            $message = xml_error_string(xml_get_error_code($this->xmlParser));
-            throw new Exception(
-                'XML error: "' . $message . '"',
-                xml_get_current_line_number($this->xmlParser),
-                xml_get_current_column_number($this->xmlParser)
-            );
+
+        $resource = fopen('data://text/plain,' . $data, 'r');
+
+        while ($data = fread($resource, 1024 * 1024)) {
```

Comment:
You said you test it on a 150 MB xml, but is it a one line xml or a multiline one?  If this is a one line (or that contains lines bigger than 1024 x 1024 characters, does it work fine?